Content.Server/GameObjects/EntitySystems/JobQueues/Job.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using Robust.Shared.Timing;
+
+namespace Content.Server.GameObjects.EntitySystems.JobQueues
+{
+    public abstract class Job<T> : IJob
```

Comment:
I'm definitely interested in getting this to use async/await to be a bit saner. That said it will require moving it to the engine to be able to set the synchronization contexts up correctly.

Reply:
Pjb did all the work on this